The event scope of a timer start event is not removed when already triggered

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->

https://github.com/camunda-cloud/zeebe/issues/6955  describes the issue that we not delete the event scope when we cancel a timer, via a new deployment. But we have also the issue that we not delete the event scope if the timer is already triggered. If the timers is not recurring or we already reached the repetitions then we will not cancel the timer, which will cause that we also not delete the event scope.

**To Reproduce**


```java
  @Test
  public void testProcessWithTriggerTimerStartEvent() {
    // given
    final var deployment =
        engineRule
            .deployment()
            .withXmlResource(
                Bpmn.createExecutableProcess(PROCESS_ID)
                    .startEvent()
                    .timerWithDate("=now() + duration(\"PT15S\")")
                    .endEvent()
                    .done())
            .deploy();

    final var processDefinitionKey =
        deployment.getValue().getProcessesMetadata().get(0).getProcessDefinitionKey();

    // when
    engineRule.awaitProcessingOf(
        RecordingExporter.timerRecords(TimerIntent.CREATED)
            .withProcessDefinitionKey(processDefinitionKey)
            .getFirst());

    engineRule.increaseTime(Duration.ofSeconds(15));

    RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_COMPLETED)
        .withProcessDefinitionKey(processDefinitionKey)
        .withElementType(BpmnElementType.PROCESS)
        .await();

    // then
    assertThatStateIsEmpty();
  }
```

<!--
Steps to reproduce the behavior

If possible add a minimal reproducer code sample
 - when using the Java client: https://github.com/zeebe-io/zeebe-test-template-java

-->

**Expected behavior**

After we triggered the timer the defined repetitions count we remove the event scope.
<!-- A clear and concise description of what you expected to happen. -->

**Log/Stacktrace**
<!-- If possible add the full stacktrace or Zeebe log which contains the issue. -->

<details><summary>Full Stacktrace</summary>
 <p>

```
/usr/lib/jvm/java-11-openjdk/bin/java -ea -Didea.test.cyclic.buffer.size=1048576 -javaagent:/opt/intellij/idea-IU-203.6682.168/lib/idea_rt.jar=34387:/opt/intellij/idea-IU-203.6682.168/bin -Dfile.encoding=UTF-8 -classpath /opt/intellij/idea-IU-203.6682.168/lib/idea_rt.jar:/opt/intellij/idea-IU-203.6682.168/plugins/junit/lib/junit5-rt.jar:/opt/intellij/idea-IU-203.6682.168/plugins/junit/lib/junit-rt.jar:/home/zell/goPath/src/github.com/zeebe-io/zeebe/engine/target/test-classes:/home/zell/goPath/src/github.com/zeebe-io/zeebe/engine/target/classes:/home/zell/goPath/src/github.com/zeebe-io/zeebe/util/target/classes:/home/zell/.m2/repository/org/apache/logging/log4j/log4j-api/2.14.1/log4j-api-2.14.1.jar:/home/zell/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.12.3/jackson-databind-2.12.3.jar:/home/zell/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.12.3/jackson-core-2.12.3.jar:/home/zell/.m2/repository/com/fasterxml/jackson/datatype/jackson-datatype-jsr310/2.12.3/jackson-datatype-jsr310-2.12.3.jar:/home/zell/.m2/repository/org/springframework/spring-core/5.3.6/spring-core-5.3.6.jar:/home/zell/.m2/repository/org/springframework/spring-jcl/5.3.6/spring-jcl-5.3.6.jar:/home/zell/.m2/repository/org/springframework/spring-context/5.3.6/spring-context-5.3.6.jar:/home/zell/.m2/repository/org/springframework/spring-aop/5.3.6/spring-aop-5.3.6.jar:/home/zell/.m2/repository/org/springframework/spring-expression/5.3.6/spring-expression-5.3.6.jar:/home/zell/.m2/repository/org/springframework/boot/spring-boot/2.4.5/spring-boot-2.4.5.jar:/home/zell/.m2/repository/org/springframework/boot/spring-boot-actuator/2.4.5/spring-boot-actuator-2.4.5.jar:/home/zell/.m2/repository/org/springframework/boot/spring-boot-autoconfigure/2.4.5/spring-boot-autoconfigure-2.4.5.jar:/home/zell/.m2/repository/org/springframework/boot/spring-boot-actuator-autoconfigure/2.4.5/spring-boot-actuator-autoconfigure-2.4.5.jar:/home/zell/.m2/repository/org/agrona/agrona/1.10.0/agrona-1.10.0.jar:/home/zell/.m2/repository/org/slf4j/slf4j-api/1.7.30/slf4j-api-1.7.30.jar:/home/zell/goPath/src/github.com/zeebe-io/zeebe/protocol/target/classes:/home/zell/goPath/src/github.com/zeebe-io/zeebe/zb-db/target/classes:/home/zell/.m2/repository/org/rocksdb/rocksdbjni/6.19.3/rocksdbjni-6.19.3.jar:/home/zell/goPath/src/github.com/zeebe-io/zeebe/protocol-impl/target/classes:/home/zell/.m2/repository/org/msgpack/jackson-dataformat-msgpack/0.8.22/jackson-dataformat-msgpack-0.8.22.jar:/home/zell/goPath/src/github.com/zeebe-io/zeebe/logstreams/target/classes:/home/zell/goPath/src/github.com/zeebe-io/zeebe/dispatcher/target/classes:/home/zell/.m2/repository/com/netflix/concurrency-limits/concurrency-limits-core/0.3.6/concurrency-limits-core-0.3.6.jar:/home/zell/goPath/src/github.com/zeebe-io/zeebe/atomix/cluster/target/classes:/home/zell/goPath/src/github.com/zeebe-io/zeebe/atomix/utils/target/classes:/home/zell/.m2/repository/com/esotericsoftware/minlog/1.3.1/minlog-1.3.1.jar:/home/zell/goPath/src/github.com/zeebe-io/zeebe/snapshot/target/classes:/home/zell/goPath/src/github.com/zeebe-io/zeebe/journal/target/classes:/home/zell/.m2/repository/io/netty/netty-transport/4.1.63.Final/netty-transport-4.1.63.Final.jar:/home/zell/.m2/repository/io/netty/netty-resolver/4.1.63.Final/netty-resolver-4.1.63.Final.jar:/home/zell/.m2/repository/io/netty/netty-codec/4.1.63.Final/netty-codec-4.1.63.Final.jar:/home/zell/.m2/repository/io/netty/netty-common/4.1.63.Final/netty-common-4.1.63.Final.jar:/home/zell/.m2/repository/io/netty/netty-transport-native-epoll/4.1.63.Final/netty-transport-native-epoll-4.1.63.Final-linux-x86_64.jar:/home/zell/.m2/repository/io/netty/netty-transport-native-unix-common/4.1.63.Final/netty-transport-native-unix-common-4.1.63.Final.jar:/home/zell/.m2/repository/org/apache/commons/commons-math3/3.6.1/commons-math3-3.6.1.jar:/home/zell/.m2/repository/com/esotericsoftware/kryo/5.1.1/kryo-5.1.1.jar:/home/zell/.m2/repository/com/esotericsoftware/reflectasm/1.11.9/reflectasm-1.11.9.jar:/home/zell/.m2/repository/io/netty/netty-buffer/4.1.63.Final/netty-buffer-4.1.63.Final.jar:/home/zell/.m2/repository/com/google/guava/guava/30.1.1-jre/guava-30.1.1-jre.jar:/home/zell/.m2/repository/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar:/home/zell/.m2/repository/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar:/home/zell/.m2/repository/org/checkerframework/checker-qual/3.8.0/checker-qual-3.8.0.jar:/home/zell/.m2/repository/com/google/j2objc/j2objc-annotations/1.3/j2objc-annotations-1.3.jar:/home/zell/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.12.3/jackson-annotations-2.12.3.jar:/home/zell/goPath/src/github.com/zeebe-io/zeebe/msgpack-core/target/classes:/home/zell/goPath/src/github.com/zeebe-io/zeebe/expression-language/target/classes:/home/zell/.m2/repository/org/camunda/feel/feel-engine/1.13.2/feel-engine-1.13.2.jar:/home/zell/.m2/repository/com/lihaoyi/fastparse_2.13/2.3.1/fastparse_2.13-2.3.1.jar:/home/zell/.m2/repository/com/lihaoyi/sourcecode_2.13/0.2.3/sourcecode_2.13-0.2.3.jar:/home/zell/.m2/repository/com/lihaoyi/geny_2.13/0.6.5/geny_2.13-0.6.5.jar:/home/zell/.m2/repository/org/scala-lang/scala-library/2.13.5/scala-library-2.13.5.jar:/home/zell/goPath/src/github.com/zeebe-io/zeebe/msgpack-value/target/classes:/home/zell/goPath/src/github.com/zeebe-io/zeebe/bpmn-model/target/classes:/home/zell/.m2/repository/org/camunda/bpm/model/camunda-xml-model/7.15.0/camunda-xml-model-7.15.0.jar:/home/zell/.m2/repository/uk/co/real-logic/sbe-tool/1.22.0/sbe-tool-1.22.0.jar:/home/zell/.m2/repository/io/prometheus/simpleclient/0.10.0/simpleclient-0.10.0.jar:/home/zell/goPath/src/github.com/zeebe-io/zeebe/protocol-asserts/target/classes:/home/zell/.m2/repository/javax/annotation/javax.annotation-api/1.3.2/javax.annotation-api-1.3.2.jar:/home/zell/.m2/repository/junit/junit/4.13.2/junit-4.13.2.jar:/home/zell/.m2/repository/org/hamcrest/hamcrest-core/2.2/hamcrest-core-2.2.jar:/home/zell/.m2/repository/org/junit/jupiter/junit-jupiter-params/5.7.1/junit-jupiter-params-5.7.1.jar:/home/zell/.m2/repository/org/apiguardian/apiguardian-api/1.1.0/apiguardian-api-1.1.0.jar:/home/zell/.m2/repository/org/junit/jupiter/junit-jupiter-api/5.7.1/junit-jupiter-api-5.7.1.jar:/home/zell/.m2/repository/org/opentest4j/opentest4j/1.2.0/opentest4j-1.2.0.jar:/home/zell/.m2/repository/org/junit/platform/junit-platform-commons/1.7.1/junit-platform-commons-1.7.1.jar:/home/zell/.m2/repository/org/assertj/assertj-core/3.19.0/assertj-core-3.19.0.jar:/home/zell/.m2/repository/org/msgpack/msgpack-core/0.8.22/msgpack-core-0.8.22.jar:/home/zell/goPath/src/github.com/zeebe-io/zeebe/test-util/target/classes:/home/zell/goPath/src/github.com/zeebe-io/zeebe/exporter-api/target/classes:/home/zell/goPath/src/github.com/zeebe-io/zeebe/clients/java/target/classes:/home/zell/goPath/src/github.com/zeebe-io/zeebe/gateway-protocol-impl/target/classes:/home/zell/goPath/src/github.com/zeebe-io/zeebe/gateway-protocol/target/classes:/home/zell/.m2/repository/io/grpc/grpc-protobuf/1.37.0/grpc-protobuf-1.37.0.jar:/home/zell/.m2/repository/com/google/api/grpc/proto-google-common-protos/2.1.0/proto-google-common-protos-2.1.0.jar:/home/zell/.m2/repository/io/grpc/grpc-protobuf-lite/1.37.0/grpc-protobuf-lite-1.37.0.jar:/home/zell/.m2/repository/io/grpc/grpc-stub/1.37.0/grpc-stub-1.37.0.jar:/home/zell/.m2/repository/io/grpc/grpc-core/1.37.0/grpc-core-1.37.0.jar:/home/zell/.m2/repository/com/google/code/gson/gson/2.8.6/gson-2.8.6.jar:/home/zell/.m2/repository/com/google/android/annotations/4.1.1.4/annotations-4.1.1.4.jar:/home/zell/.m2/repository/io/perfmark/perfmark-api/0.23.0/perfmark-api-0.23.0.jar:/home/zell/.m2/repository/io/grpc/grpc-netty/1.37.0/grpc-netty-1.37.0.jar:/home/zell/.m2/repository/io/netty/netty-codec-http2/4.1.63.Final/netty-codec-http2-4.1.63.Final.jar:/home/zell/.m2/repository/io/netty/netty-codec-http/4.1.63.Final/netty-codec-http-4.1.63.Final.jar:/home/zell/.m2/repository/io/netty/netty-handler-proxy/4.1.63.Final/netty-handler-proxy-4.1.63.Final.jar:/home/zell/.m2/repository/io/netty/netty-codec-socks/4.1.63.Final/netty-codec-socks-4.1.63.Final.jar:/home/zell/.m2/repository/com/google/protobuf/protobuf-java/3.15.8/protobuf-java-3.15.8.jar:/home/zell/.m2/repository/io/netty/netty-tcnative-boringssl-static/2.0.39.Final/netty-tcnative-boringssl-static-2.0.39.Final.jar:/home/zell/.m2/repository/io/netty/netty-handler/4.1.63.Final/netty-handler-4.1.63.Final.jar:/home/zell/.m2/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-yaml/2.12.3/jackson-dataformat-yaml-2.12.3.jar:/home/zell/.m2/repository/org/testcontainers/testcontainers/1.15.3/testcontainers-1.15.3.jar:/home/zell/.m2/repository/org/apache/commons/commons-compress/1.20/commons-compress-1.20.jar:/home/zell/.m2/repository/org/rnorth/duct-tape/duct-tape/1.0.8/duct-tape-1.0.8.jar:/home/zell/.m2/repository/org/rnorth/visible-assertions/visible-assertions/2.1.2/visible-assertions-2.1.2.jar:/home/zell/.m2/repository/net/java/dev/jna/jna/5.8.0/jna-5.8.0.jar:/home/zell/.m2/repository/com/github/docker-java/docker-java-api/3.2.8/docker-java-api-3.2.8.jar:/home/zell/.m2/repository/com/github/docker-java/docker-java-transport-zerodep/3.2.8/docker-java-transport-zerodep-3.2.8.jar:/home/zell/.m2/repository/com/github/docker-java/docker-java-transport/3.2.8/docker-java-transport-3.2.8.jar:/home/zell/.m2/repository/org/springframework/spring-beans/5.3.6/spring-beans-5.3.6.jar:/home/zell/.m2/repository/org/yaml/snakeyaml/1.28/snakeyaml-1.28.jar:/home/zell/.m2/repository/io/grpc/grpc-api/1.37.0/grpc-api-1.37.0.jar:/home/zell/.m2/repository/io/grpc/grpc-context/1.37.0/grpc-context-1.37.0.jar:/home/zell/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar:/home/zell/.m2/repository/com/google/errorprone/error_prone_annotations/2.6.0/error_prone_annotations-2.6.0.jar:/home/zell/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.20/animal-sniffer-annotations-1.20.jar:/home/zell/.m2/repository/org/mockito/mockito-core/3.9.0/mockito-core-3.9.0.jar:/home/zell/.m2/repository/net/bytebuddy/byte-buddy/1.10.20/byte-buddy-1.10.20.jar:/home/zell/.m2/repository/net/bytebuddy/byte-buddy-agent/1.10.20/byte-buddy-agent-1.10.20.jar:/home/zell/.m2/repository/org/objenesis/objenesis/3.2/objenesis-3.2.jar:/home/zell/.m2/repository/org/awaitility/awaitility/4.0.3/awaitility-4.0.3.jar:/home/zell/.m2/repository/org/hamcrest/hamcrest/2.2/hamcrest-2.2.jar:/home/zell/.m2/repository/org/apache/commons/commons-lang3/3.12.0/commons-lang3-3.12.0.jar:/home/zell/goPath/src/github.com/zeebe-io/zeebe/logstreams/target/test-classes:/home/zell/goPath/src/github.com/zeebe-io/zeebe/build-tools/target/classes:/home/zell/.m2/repository/org/apache/logging/log4j/log4j-core/2.14.1/log4j-core-2.14.1.jar:/home/zell/.m2/repository/org/apache/logging/log4j/log4j-slf4j-impl/2.14.1/log4j-slf4j-impl-2.14.1.jar com.intellij.rt.junit.JUnitStarter -ideVersion5 -junit4 io.camunda.zeebe.engine.state.ProcessExecutionCleanStateTest,testProcessWithTriggerTimerStartEvent
07:39:57.473 [Broker-0-LogStream-1] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
07:39:57.637 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.logstreams - Recovering state of partition 1 from snapshot
07:39:57.751 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@77fb4a7b)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@6080bf4f, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@6605bc69, configuration: Configuration(false)]
07:39:57.861 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@756a1cf5)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@58ad48ea, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@813bd8a, configuration: Configuration(false)]
07:39:57.875 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.logstreams - Recovered state of partition 1 from snapshot at position -1
07:39:57.887 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@2115ea3e)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@6a9ccb8a, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@6d31aaa5, configuration: Configuration(false)]
07:39:57.895 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@2163816c)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@326d8c2d, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@593ca943, configuration: Configuration(false)]
07:39:58.749 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@2f2fa178)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@b65ee4b, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@72182236, configuration: Configuration(false)]
07:40:09.009 [] INFO  io.camunda.zeebe.test - Test failed, following records were exported:
07:40:09.160 [] INFO  io.camunda.zeebe.test - Compact log representation:
--------
	['C'ommand/'E'event/'R'ejection] [valueType] [intent] - #[position]->#[source record position]  P[partitionId]K[key] - [summary of value]
	P9K999 - key; #999 - record position; "ID" element/process id; @"elementid"/[P9K999] - element with ID and key
	Keys are decomposed into partition id and per partition key (e.g. 2251799813685253 -> P1K005). If single partition, the partition is omitted.
	Long IDs are shortened (e.g. 'startEvent_5d56488e-0570-416c-ba2d-36d2a3acea78' -> 'star..acea78'
--------
C DEPLOYMENT CREATE         - #01-> -1  -1 - 
E PROC       CREATED        - #02->#01 K01 - process.xml -> "process" (version:1)
E TIMER      CREATED        - #03->#01 K03 -  @"startEv..68bbb01"[ -1]  in <process "K01"[?]> due T07:40:13.805
E DEPLOYMENT CREATED        - #04->#01 K02 - process.xml
E DEPLOYMENT FULLY_DISTR    - #05->#01 K02 - 
C TIMER      TRIGGER        - #06->#01 K03 -  @"startEv..68bbb01"[ -1]  in <process "K01"[?]> due T07:40:13.805
E TIMER      TRIGGERED      - #07->#06 K03 -  @"startEv..68bbb01"[ -1]  in <process "K01"[?]> due T07:40:13.805
E PROC_EVNT  TRIGGERING     - #08->#06 K05 -  @"startEv..68bbb01"[K01] in <process K01[K04]> (no vars)
C PROC_INST  ACTIVATE       - #09->#06 K04 - PROCESS "process" in <process "process"[K04]>
E PROC_INST  ACTIVATING     - #10->#09 K04 - PROCESS "process" in <process "process"[K04]>
E PROC_INST  ACTIVATED      - #11->#09 K04 - PROCESS "process" in <process "process"[K04]>
E PROC_EVNT  TRIGGERED      - #12->#09 K05 -  @"startEv..68bbb01"[K01] in <process K01[K04]> (no vars)
C PROC_INST  ACTIVATE       - #13->#09 K06 - START_EVENT "startEv..68bbb01" in <process "process"[K04]>
E PROC_INST  ACTIVATING     - #14->#13 K06 - START_EVENT "startEv..68bbb01" in <process "process"[K04]>
E PROC_INST  ACTIVATED      - #15->#13 K06 - START_EVENT "startEv..68bbb01" in <process "process"[K04]>
C PROC_INST  COMPLETE       - #16->#13 K06 - START_EVENT "startEv..68bbb01" in <process "process"[K04]>
E PROC_INST  COMPLETING     - #17->#16 K06 - START_EVENT "startEv..68bbb01" in <process "process"[K04]>
E PROC_INST  COMPLETED      - #18->#16 K06 - START_EVENT "startEv..68bbb01" in <process "process"[K04]>
E PROC_INST  SEQ_FLOW_TAKEN - #19->#16 K07 - SEQUENCE_FLOW "sequenc..080d5ce" in <process "process"[K04]>
C PROC_INST  ACTIVATE       - #20->#16 K08 - END_EVENT "endEven..3dd6a82" in <process "process"[K04]>
E PROC_INST  ACTIVATING     - #21->#20 K08 - END_EVENT "endEven..3dd6a82" in <process "process"[K04]>
E PROC_INST  ACTIVATED      - #22->#20 K08 - END_EVENT "endEven..3dd6a82" in <process "process"[K04]>
E PROC_INST  COMPLETING     - #23->#20 K08 - END_EVENT "endEven..3dd6a82" in <process "process"[K04]>
E PROC_INST  COMPLETED      - #24->#20 K08 - END_EVENT "endEven..3dd6a82" in <process "process"[K04]>
C PROC_INST  COMPLETE       - #25->#20 K04 - PROCESS "process" in <process "process"[K04]>
E PROC_INST  COMPLETING     - #26->#25 K04 - PROCESS "process" in <process "process"[K04]>
E PROC_INST  COMPLETED      - #27->#25 K04 - PROCESS "process" in <process "process"[K04]>

-------------- Deployed Processes ----------------------
process.xml -> "process" (version:1)[K01] ------
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<definitions xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" id="definitions_acd69e66-9373-48a1-96ee-dd6c57d2ccab" targetNamespace="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL">
  <process id="process" isExecutable="true">
    <startEvent id="startEvent_68303000-b6c0-49d9-a1b3-9516168bbb01">
      <outgoing>sequenceFlow_28516602-ed14-4256-8303-3a2d9080d5ce</outgoing>
      <timerEventDefinition id="timerEventDefinition_ab340773-6401-404b-aa83-7b8d7f8499c7">
        <timeDate id="timeDate_f058122d-26d7-4e39-ae0d-d2245a428e61">=now() + duration("PT15S")</timeDate>
      </timerEventDefinition>
    </startEvent>
    <endEvent id="endEvent_46ab3e36-fdbf-40a5-a01e-e7fb33dd6a82">
      <incoming>sequenceFlow_28516602-ed14-4256-8303-3a2d9080d5ce</incoming>
    </endEvent>
    <sequenceFlow id="sequenceFlow_28516602-ed14-4256-8303-3a2d9080d5ce" sourceRef="startEvent_68303000-b6c0-49d9-a1b3-9516168bbb01" targetRef="endEvent_46ab3e36-fdbf-40a5-a01e-e7fb33dd6a82"/>
  </process>
  <bpmndi:BPMNDiagram id="BPMNDiagram_2b208eeb-af41-40d4-ad8e-59ce26a7cbd3">
    <bpmndi:BPMNPlane bpmnElement="process" id="BPMNPlane_8b324c3c-d4b7-4087-a774-15222eaef857">
      <bpmndi:BPMNShape bpmnElement="startEvent_68303000-b6c0-49d9-a1b3-9516168bbb01" id="BPMNShape_d67f8c14-cbf5-4704-9636-675b01d6c6ec">
        <dc:Bounds height="36.0" width="36.0" x="100.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="endEvent_46ab3e36-fdbf-40a5-a01e-e7fb33dd6a82" id="BPMNShape_e384b91e-55f1-44a5-97c6-b01a45f41503">
        <dc:Bounds height="36.0" width="36.0" x="186.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_28516602-ed14-4256-8303-3a2d9080d5ce" id="BPMNEdge_37dd86e6-a407-43b6-b3e8-959ee0728a35">
        <di:waypoint x="136.0" y="118.0"/>
        <di:waypoint x="186.0" y="118.0"/>
      </bpmndi:BPMNEdge>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</definitions>


--------------- Decomposed keys (for debugging) -----------------
 -1 <-> -1
K01 <-> 2251799813685249
K02 <-> 2251799813685250
K03 <-> 2251799813685251
K04 <-> 2251799813685252
K05 <-> 2251799813685253
K06 <-> 2251799813685254
K07 <-> 2251799813685255
K08 <-> 2251799813685256

07:40:09.162 [] DEBUG io.camunda.zeebe.util.buffer - Close stream processor
07:40:09.163 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.logstreams - Closed stream processor controller Broker-0-StreamProcessor-1.
07:40:09.198 [Broker-0-LogStream-1] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream-1
07:40:09.199 [stream-1-write-buffer] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
07:40:09.199 [Broker-0-LogStream-1] INFO  io.camunda.zeebe.logstreams - On closing logstream stream-1 close 2 readers
07:40:09.200 [] DEBUG io.camunda.zeebe.broker.test - Clean up test files on path /tmp/junit17066309036761455744
07:40:09.200 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
07:40:09.201 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
07:40:09.201 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
07:40:09.202 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully

org.awaitility.core.ConditionTimeoutException: Assertion condition defined as a lambda expression in io.camunda.zeebe.engine.state.ProcessExecutionCleanStateTest [Expected all columns to be empty] 
Expecting empty but was: [EVENT_SCOPE] within 10 seconds.

	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:679)
	at io.camunda.zeebe.engine.state.ProcessExecutionCleanStateTest.assertThatStateIsEmpty(ProcessExecutionCleanStateTest.java:719)
	at io.camunda.zeebe.engine.state.ProcessExecutionCleanStateTest.testProcessWithTriggerTimerStartEvent(ProcessExecutionCleanStateTest.java:663)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:220)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:53)
Caused by: java.lang.AssertionError: [Expected all columns to be empty] 
Expecting empty but was: [EVENT_SCOPE]
	at io.camunda.zeebe.engine.state.ProcessExecutionCleanStateTest.lambda$assertThatStateIsEmpty$18(ProcessExecutionCleanStateTest.java:727)
	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:222)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:209)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)


Process finished with exit code 255
```

</p>
</details>


**Environment:**
 - OS: <!-- [e.g. Linux] -->
 - Zeebe Version:SNAPSHOT / RC1
 - Configuration: <!-- [e.g. exporters etc.] -->

--- a/engine/src/main/java/io/camunda/zeebe/engine/processing/timer/TriggerTimerProcessor.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/processing/timer/TriggerTimerProcessor.java
@@ -104,11 +104,6 @@ public final class TriggerTimerProcessor implements TypedRecordProcessor<TimerRe
         processState.getFlowElement(
             processDefinitionKey, timer.getTargetElementIdBuffer(), ExecutableCatchEvent.class);
     if (isStartEvent(elementInstanceKey)) {
-      if (!eventScopeInstanceState.isAcceptingEvent(processDefinitionKey)) {
-        rejectNoActiveTimer(record);
-        return;
-      }
-
       stateWriter.appendFollowUpEvent(record.getKey(), TimerIntent.TRIGGERED, timer);
       final long processInstanceKey = keyGenerator.nextKey();
       eventHandle.activateProcessInstanceForStartEvent(

--- a/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/EventAppliers.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/EventAppliers.java
@@ -175,7 +175,7 @@ public final class EventAppliers implements EventApplier {
     register(
         MessageStartEventSubscriptionIntent.CREATED,
         new MessageStartEventSubscriptionCreatedApplier(
-            state.getMessageStartEventSubscriptionState(), state.getEventScopeInstanceState()));
+            state.getMessageStartEventSubscriptionState()));
     register(
         MessageStartEventSubscriptionIntent.CORRELATED,
         new MessageStartEventSubscriptionCorrelatedApplier(
@@ -183,7 +183,7 @@ public final class EventAppliers implements EventApplier {
     register(
         MessageStartEventSubscriptionIntent.DELETED,
         new MessageStartEventSubscriptionDeletedApplier(
-            state.getMessageStartEventSubscriptionState(), state.getEventScopeInstanceState()));
+            state.getMessageStartEventSubscriptionState()));
   }
 
   private void registerIncidentEventAppliers(final MutableZeebeState state) {

--- a/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/MessageStartEventSubscriptionCorrelatedApplier.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/MessageStartEventSubscriptionCorrelatedApplier.java
@@ -43,7 +43,7 @@ public final class MessageStartEventSubscriptionCorrelatedApplier
     }
 
     // write the event trigger for the start event
-    eventScopeInstanceState.triggerEvent(
+    eventScopeInstanceState.triggerStartEvent(
         value.getProcessDefinitionKey(),
         value.getMessageKey(),
         value.getStartEventIdBuffer(),

--- a/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/MessageStartEventSubscriptionCreatedApplier.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/MessageStartEventSubscriptionCreatedApplier.java
@@ -8,36 +8,23 @@
 package io.camunda.zeebe.engine.state.appliers;
 
 import io.camunda.zeebe.engine.state.TypedEventApplier;
-import io.camunda.zeebe.engine.state.mutable.MutableEventScopeInstanceState;
 import io.camunda.zeebe.engine.state.mutable.MutableMessageStartEventSubscriptionState;
 import io.camunda.zeebe.protocol.impl.record.value.message.MessageStartEventSubscriptionRecord;
 import io.camunda.zeebe.protocol.record.intent.MessageStartEventSubscriptionIntent;
-import java.util.Collections;
-import java.util.List;
-import org.agrona.DirectBuffer;
 
 public final class MessageStartEventSubscriptionCreatedApplier
     implements TypedEventApplier<
         MessageStartEventSubscriptionIntent, MessageStartEventSubscriptionRecord> {
 
-  private static final List<DirectBuffer> NO_INTERRUPTING_ELEMENT_IDS = Collections.emptyList();
-
   private final MutableMessageStartEventSubscriptionState subscriptionState;
-  private final MutableEventScopeInstanceState eventScopeInstanceState;
 
   public MessageStartEventSubscriptionCreatedApplier(
-      final MutableMessageStartEventSubscriptionState subscriptionState,
-      final MutableEventScopeInstanceState eventScopeInstanceState) {
+      final MutableMessageStartEventSubscriptionState subscriptionState) {
     this.subscriptionState = subscriptionState;
-    this.eventScopeInstanceState = eventScopeInstanceState;
   }
 
   @Override
   public void applyState(final long key, final MessageStartEventSubscriptionRecord value) {
-
     subscriptionState.put(key, value);
-
-    eventScopeInstanceState.createIfNotExists(
-        value.getProcessDefinitionKey(), NO_INTERRUPTING_ELEMENT_IDS);
   }
 }

--- a/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/MessageStartEventSubscriptionDeletedApplier.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/MessageStartEventSubscriptionDeletedApplier.java
@@ -8,7 +8,6 @@
 package io.camunda.zeebe.engine.state.appliers;
 
 import io.camunda.zeebe.engine.state.TypedEventApplier;
-import io.camunda.zeebe.engine.state.mutable.MutableEventScopeInstanceState;
 import io.camunda.zeebe.engine.state.mutable.MutableMessageStartEventSubscriptionState;
 import io.camunda.zeebe.protocol.impl.record.value.message.MessageStartEventSubscriptionRecord;
 import io.camunda.zeebe.protocol.record.intent.MessageStartEventSubscriptionIntent;
@@ -18,21 +17,15 @@ public final class MessageStartEventSubscriptionDeletedApplier
         MessageStartEventSubscriptionIntent, MessageStartEventSubscriptionRecord> {
 
   private final MutableMessageStartEventSubscriptionState subscriptionState;
-  private final MutableEventScopeInstanceState eventScopeInstanceState;
 
   public MessageStartEventSubscriptionDeletedApplier(
-      final MutableMessageStartEventSubscriptionState subscriptionState,
-      final MutableEventScopeInstanceState eventScopeInstanceState) {
+      final MutableMessageStartEventSubscriptionState subscriptionState) {
     this.subscriptionState = subscriptionState;
-    this.eventScopeInstanceState = eventScopeInstanceState;
   }
 
   @Override
   public void applyState(final long key, final MessageStartEventSubscriptionRecord value) {
     final var processDefinitionKey = value.getProcessDefinitionKey();
-
     subscriptionState.remove(processDefinitionKey, value.getMessageNameBuffer());
-
-    eventScopeInstanceState.deleteInstance(processDefinitionKey);
   }
 }

--- a/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ProcessCreatedApplier.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ProcessCreatedApplier.java
@@ -7,36 +7,22 @@
  */
 package io.camunda.zeebe.engine.state.appliers;
 
-import io.camunda.zeebe.engine.processing.deployment.model.element.ExecutableCatchEventElement;
 import io.camunda.zeebe.engine.state.TypedEventApplier;
-import io.camunda.zeebe.engine.state.mutable.MutableEventScopeInstanceState;
 import io.camunda.zeebe.engine.state.mutable.MutableProcessState;
 import io.camunda.zeebe.engine.state.mutable.MutableZeebeState;
 import io.camunda.zeebe.protocol.impl.record.value.deployment.ProcessRecord;
 import io.camunda.zeebe.protocol.record.intent.ProcessIntent;
-import java.util.Collections;
 
 public class ProcessCreatedApplier implements TypedEventApplier<ProcessIntent, ProcessRecord> {
 
   private final MutableProcessState processState;
-  private final MutableEventScopeInstanceState eventScopeInstanceState;
 
   public ProcessCreatedApplier(final MutableZeebeState state) {
     processState = state.getProcessState();
-    eventScopeInstanceState = state.getEventScopeInstanceState();
   }
 
   @Override
   public void applyState(final long processDefinitionKey, final ProcessRecord value) {
     processState.putProcess(processDefinitionKey, value);
-
-    // timer start events
-    final var hasAtLeastOneTimer =
-        processState.getProcessByKey(processDefinitionKey).getProcess().getStartEvents().stream()
-            .anyMatch(ExecutableCatchEventElement::isTimer);
-
-    if (hasAtLeastOneTimer) {
-      eventScopeInstanceState.createIfNotExists(processDefinitionKey, Collections.emptyList());
-    }
   }
 }

--- a/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ProcessEventTriggeringApplier.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ProcessEventTriggeringApplier.java
@@ -43,7 +43,11 @@ final class ProcessEventTriggeringApplier
     final var targetElementIdBuffer = value.getTargetElementIdBuffer();
     final var scopeKey = value.getScopeKey();
 
-    eventScopeState.triggerEvent(scopeKey, key, targetElementIdBuffer, variables);
+    if (value.getProcessDefinitionKey() == scopeKey) {
+      eventScopeState.triggerStartEvent(scopeKey, key, targetElementIdBuffer, variables);
+    } else {
+      eventScopeState.triggerEvent(scopeKey, key, targetElementIdBuffer, variables);
+    }
     eventSubProcessInterruptionMarker.markInstanceIfInterrupted(
         scopeKey, value.getProcessDefinitionKey(), targetElementIdBuffer);
   }

--- a/engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbEventScopeInstanceState.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbEventScopeInstanceState.java
@@ -137,6 +137,15 @@ public final class DbEventScopeInstanceState implements MutableEventScopeInstanc
   }
 
   @Override
+  public void triggerStartEvent(
+      final long processDefinitionKey,
+      final long eventKey,
+      final DirectBuffer elementId,
+      final DirectBuffer variables) {
+    createTrigger(processDefinitionKey, eventKey, elementId, variables);
+  }
+
+  @Override
   public void deleteTrigger(final long eventScopeKey, final long eventKey) {
     eventTriggerScopeKey.wrapLong(eventScopeKey);
     eventTriggerEventKey.wrapLong(eventKey);

--- a/engine/src/main/java/io/camunda/zeebe/engine/state/mutable/MutableEventScopeInstanceState.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/state/mutable/MutableEventScopeInstanceState.java
@@ -67,12 +67,13 @@ public interface MutableEventScopeInstanceState extends EventScopeInstanceState 
       long eventScopeKey, long eventKey, DirectBuffer elementId, DirectBuffer variables);
 
   /**
-   * Creates an eventTrigger for an start event as scope key the process definition key is used.
+   * Creates an event trigger for a process start event. Uses the process definition key as the
+   * scope key of the trigger.
    *
-   * @param processDefinitionKey the process definition key of the to be triggered start event, this
-   *     is used as scope key for the event trigger
+   * @param processDefinitionKey the key of the process definition a new instance should be created
+   *     of
    * @param eventKey the key of the event record (used for ordering)
-   * @param elementId the id of the element which should be triggered
+   * @param elementId the id of the start event which should be triggered
    * @param variables the variables of the occurred event, i.e. message variables
    */
   void triggerStartEvent(

--- a/engine/src/main/java/io/camunda/zeebe/engine/state/mutable/MutableEventScopeInstanceState.java
+++ b/engine/src/main/java/io/camunda/zeebe/engine/state/mutable/MutableEventScopeInstanceState.java
@@ -67,6 +67,18 @@ public interface MutableEventScopeInstanceState extends EventScopeInstanceState 
       long eventScopeKey, long eventKey, DirectBuffer elementId, DirectBuffer variables);
 
   /**
+   * Creates an eventTrigger for an start event as scope key the process definition key is used.
+   *
+   * @param processDefinitionKey the process definition key of the to be triggered start event, this
+   *     is used as scope key for the event trigger
+   * @param eventKey the key of the event record (used for ordering)
+   * @param elementId the id of the element which should be triggered
+   * @param variables the variables of the occurred event, i.e. message variables
+   */
+  void triggerStartEvent(
+      long processDefinitionKey, long eventKey, DirectBuffer elementId, DirectBuffer variables);
+
+  /**
    * Deletes an event trigger by key and scope key. Does not fail if the trigger does not exist.
    *
    * @param eventScopeKey the key of the event scope

--- a/engine/src/test/java/io/camunda/zeebe/engine/state/ProcessExecutionCleanStateTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/state/ProcessExecutionCleanStateTest.java
@@ -630,6 +630,39 @@ public final class ProcessExecutionCleanStateTest {
   }
 
   @Test
+  public void testProcessWithMessageStartEventAndRedeployWithout() {
+    // given
+    final var deployment =
+        engineRule
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(PROCESS_ID)
+                    .startEvent()
+                    .message(m -> m.name("msg").zeebeCorrelationKey("=123"))
+                    .endEvent()
+                    .done())
+            .deploy();
+
+    final var processDefinitionKey =
+        deployment.getValue().getProcessesMetadata().get(0).getProcessDefinitionKey();
+
+    // when
+    // deploy new process without msg start event to delete the subscription and event scope
+    engineRule
+        .deployment()
+        .withXmlResource(Bpmn.createExecutableProcess(PROCESS_ID).startEvent().endEvent().done())
+        .deploy();
+
+    RecordingExporter.messageStartEventSubscriptionRecords(
+            MessageStartEventSubscriptionIntent.DELETED)
+        .withWorkfloKey(processDefinitionKey)
+        .await();
+
+    // then
+    assertThatStateIsEmpty();
+  }
+
+  @Test
   public void testProcessWithTriggerTimerStartEvent() {
     // given
     final var deployment =

--- a/engine/src/test/java/io/camunda/zeebe/engine/state/ProcessExecutionCleanStateTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/state/ProcessExecutionCleanStateTest.java
@@ -630,6 +630,40 @@ public final class ProcessExecutionCleanStateTest {
   }
 
   @Test
+  public void testProcessWithTriggerTimerStartEvent() {
+    // given
+    final var deployment =
+        engineRule
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(PROCESS_ID)
+                    .startEvent()
+                    .timerWithDate("=now() + duration(\"PT15S\")")
+                    .endEvent()
+                    .done())
+            .deploy();
+
+    final var processDefinitionKey =
+        deployment.getValue().getProcessesMetadata().get(0).getProcessDefinitionKey();
+
+    // when
+    engineRule.awaitProcessingOf(
+        RecordingExporter.timerRecords(TimerIntent.CREATED)
+            .withProcessDefinitionKey(processDefinitionKey)
+            .getFirst());
+
+    engineRule.increaseTime(Duration.ofSeconds(15));
+
+    RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_COMPLETED)
+        .withProcessDefinitionKey(processDefinitionKey)
+        .withElementType(BpmnElementType.PROCESS)
+        .await();
+
+    // then
+    assertThatStateIsEmpty();
+  }
+
+  @Test
   public void testProcessWithTimerStartEventRedeployment() {
     // given
     final var deployment =

--- a/engine/src/test/java/io/camunda/zeebe/engine/state/ProcessExecutionCleanStateTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/state/ProcessExecutionCleanStateTest.java
@@ -18,6 +18,7 @@ import io.camunda.zeebe.protocol.record.intent.IncidentIntent;
 import io.camunda.zeebe.protocol.record.intent.JobIntent;
 import io.camunda.zeebe.protocol.record.intent.MessageStartEventSubscriptionIntent;
 import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
+import io.camunda.zeebe.protocol.record.intent.TimerIntent;
 import io.camunda.zeebe.protocol.record.value.BpmnElementType;
 import io.camunda.zeebe.test.util.record.RecordingExporter;
 import java.time.Duration;
@@ -590,6 +591,88 @@ public final class ProcessExecutionCleanStateTest {
     RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_TERMINATED)
         .withProcessInstanceKey(processInstanceKey)
         .withElementType(BpmnElementType.PROCESS)
+        .await();
+
+    // then
+    assertThatStateIsEmpty();
+  }
+
+  @Test
+  public void testProcessWithTimerStartEvent() {
+    // given
+    final var deployment =
+        engineRule
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(PROCESS_ID)
+                    .startEvent()
+                    .timerWithCycle("R/PT10S")
+                    .endEvent()
+                    .done())
+            .deploy();
+
+    final var processDefinitionKey =
+        deployment.getValue().getProcessesMetadata().get(0).getProcessDefinitionKey();
+
+    // when
+    // deploy new process without timer start event to delete the timer
+    engineRule
+        .deployment()
+        .withXmlResource(Bpmn.createExecutableProcess(PROCESS_ID).startEvent().endEvent().done())
+        .deploy();
+
+    RecordingExporter.timerRecords(TimerIntent.CANCELED)
+        .withProcessDefinitionKey(processDefinitionKey)
+        .await();
+
+    // then
+    assertThatStateIsEmpty();
+  }
+
+  @Test
+  public void testProcessWithTimerStartEventRedeployment() {
+    // given
+    final var deployment =
+        engineRule
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(PROCESS_ID)
+                    .startEvent()
+                    .timerWithCycle("R/PT10S")
+                    .endEvent()
+                    .done())
+            .deploy();
+
+    final var processDefinitionKey =
+        deployment.getValue().getProcessesMetadata().get(0).getProcessDefinitionKey();
+
+    final var deploy2 =
+        engineRule
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(PROCESS_ID)
+                    .startEvent()
+                    .timerWithCycle("R/PT5S")
+                    .endEvent()
+                    .done())
+            .deploy();
+
+    final var processDefinitionKey2 =
+        deployment.getValue().getProcessesMetadata().get(0).getProcessDefinitionKey();
+
+    // when
+    // deploy new process without timer start event to delete the timer
+    engineRule
+        .deployment()
+        .withXmlResource(Bpmn.createExecutableProcess(PROCESS_ID).startEvent().endEvent().done())
+        .deploy();
+
+    RecordingExporter.timerRecords(TimerIntent.CANCELED)
+        .withProcessDefinitionKey(processDefinitionKey)
+        .await();
+
+    RecordingExporter.timerRecords(TimerIntent.CANCELED)
+        .withProcessDefinitionKey(processDefinitionKey2)
         .await();
 
     // then

--- a/engine/src/test/java/io/camunda/zeebe/engine/state/instance/EventScopeInstanceStateTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/state/instance/EventScopeInstanceStateTest.java
@@ -9,6 +9,7 @@ package io.camunda.zeebe.engine.state.instance;
 
 import static io.camunda.zeebe.util.buffer.BufferUtil.wrapString;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
 
 import io.camunda.zeebe.engine.state.mutable.MutableEventScopeInstanceState;
 import io.camunda.zeebe.engine.state.mutable.MutableZeebeState;
@@ -149,6 +150,19 @@ public final class EventScopeInstanceStateTest {
   }
 
   @Test
+  public void shouldTriggerStartEventForNonExistingEventScope() {
+    // given
+    final long scopeKey = 123;
+    final EventTrigger eventTrigger = createEventTrigger();
+
+    // when
+    triggerStartEvent(scopeKey, 456, eventTrigger);
+
+    // then
+    Assertions.assertThat(state.peekEventTrigger(scopeKey)).isEqualTo(eventTrigger);
+  }
+
+  @Test
   public void shouldPeekEventTrigger() {
     // given
     final long key = 123;
@@ -182,6 +196,23 @@ public final class EventScopeInstanceStateTest {
   }
 
   @Test
+  public void shouldPollStartEventTrigger() {
+    // given
+    final long scopeKey = 123;
+    final EventTrigger eventTrigger1 = createEventTrigger();
+    final EventTrigger eventTrigger2 = createEventTrigger();
+
+    // when
+    triggerStartEvent(scopeKey, 1, eventTrigger1);
+    triggerStartEvent(scopeKey, 2, eventTrigger2);
+
+    // then
+    Assertions.assertThat(state.pollEventTrigger(scopeKey)).isEqualTo(eventTrigger1);
+    Assertions.assertThat(state.pollEventTrigger(scopeKey)).isEqualTo(eventTrigger2);
+    Assertions.assertThat(state.pollEventTrigger(scopeKey)).isNull();
+  }
+
+  @Test
   public void shouldDeleteTrigger() {
     // given
     final long key = 123;
@@ -195,6 +226,20 @@ public final class EventScopeInstanceStateTest {
 
     // then
     Assertions.assertThat(state.pollEventTrigger(key)).isNull();
+  }
+
+  @Test
+  public void shouldDeleteStartEventTrigger() {
+    // given
+    final long scopeKey = 123;
+    final EventTrigger eventTrigger1 = createEventTrigger();
+    triggerStartEvent(scopeKey, 1, eventTrigger1);
+
+    // when
+    state.deleteTrigger(scopeKey, 1);
+
+    // then
+    Assertions.assertThat(state.pollEventTrigger(scopeKey)).isNull();
   }
 
   @Test
@@ -229,6 +274,29 @@ public final class EventScopeInstanceStateTest {
 
     // then
     assertThat(triggered).isFalse();
+  }
+
+  @Test
+  public void shouldDeleteStartEventTriggerOnDeletionOfInstance() {
+    // given
+    final long scopeKey = 123;
+    final EventTrigger eventTrigger1 = createEventTrigger();
+    triggerStartEvent(scopeKey, 1, eventTrigger1);
+
+    // when
+    state.deleteInstance(scopeKey);
+
+    // then
+    Assertions.assertThat(state.pollEventTrigger(scopeKey)).isNull();
+  }
+
+  @Test
+  public void shouldNotFailOnDeletionOfNonExistingInstance() {
+    // given
+    final long key = 123;
+
+    // expect no exception
+    assertThatNoException().isThrownBy(() -> state.deleteInstance(key));
   }
 
   @Test
@@ -278,6 +346,12 @@ public final class EventScopeInstanceStateTest {
   private boolean triggerEvent(
       final long eventScopeKey, final long eventKey, final EventTrigger eventTrigger) {
     return state.triggerEvent(
+        eventScopeKey, eventKey, eventTrigger.getElementId(), eventTrigger.getVariables());
+  }
+
+  private void triggerStartEvent(
+      final long eventScopeKey, final long eventKey, final EventTrigger eventTrigger) {
+    state.triggerStartEvent(
         eventScopeKey, eventKey, eventTrigger.getElementId(), eventTrigger.getVariables());
   }
 

